libass/ass.c

```diff
@@ -90,6 +84,14 @@ void ass_free_track(ASS_Track *track)
             ass_free_event(track, i);
     }
     free(track->events);
+
+    if (track->parser_priv) {
+        free(track->parser_priv->read_order_bitmap);
+        free(track->parser_priv->fontname);
+        free(track->parser_priv->fontdata);
+        free(track->parser_priv);
+    }
+
```

Comment:
Could you explain what the double-free issue was and why reordering fixed it?

Reply:
`ass_free_track` does not mirror the allocations done in `ass_new_track`. The first element allocated in `ass_new_track` should be the last free'd in `ass_free_track`.

The code in the first commit assume this mirrored symmetry, and ass_free_event access `track->parser_priv->events_size`, which would not exist due to the `free(track->parser_priv);` happening too early.